src/cleanfigure.m

```diff
@@ -347,104 +349,162 @@ function simplifyLine(meta, handle, targetResolution)
+        xLim   = xlim(meta.gca);
+        yLim   = ylim(meta.gca);
+        % In 3D plots the yAxis normally runs from positive to negative
+        % values
+        yLim   = [yLim(2), yLim(1)];
+        zLim   = zlim(meta.gca);
+
+        if isXlog
+            xLim  = log10(xLim);
+        end
+        if isYlog
+            yLim  = log10(yLim);
+        end
+        if isZlog
+            zLim  = log10(zLim);
+        end
+
+        % Project the limits to 2D coordinates
+        Limits = C*[xLim(:), yLim(:), zLim(:), [1; 1]]';
+
+        % Assign the new projected 2D limits
+        xLim   = Limits(1, :);
+        yLim   = Limits(2, :);
+    else
+        data   = [xData(:),...
+                  yData(:)];
+        xLim   = xlim(meta.gca);
+        yLim   = ylim(meta.gca);
+
+        if isXlog
+            xLim  = log10(xLim);
+        end
+        if isYlog
+            yLim  = log10(yLim);
+        end
     end
-    yrange = yLim(2)-yLim(1);
+
+    % Automatically guess a tol based on the area of the figure and
+    % the area and resolution of the output
+    xRange = xLim(2)-xLim(1);
+    yRange = yLim(2)-yLim(1);
 
     % Conversion factors of data units into pixels
-    xToPix = W/xrange;
-    yToPix = H/yrange;
+    xToPix = W/xRange;
+    yToPix = H/yRange;
 
     % If the path has markers, perform pixelation instead of simplification
     hasMarkers = ~strcmpi(get(handle,'Marker'),'none');
     if hasMarkers
         % Pixelate data at the zoom multiplier
-        mask   = pixelate(vxData, vyData, xToPix, yToPix);
-        xData  = xData(mask);
-        yData  = yData(mask);
-        set(handle, 'XData', xData);
-        set(handle, 'YData', yData);
+        mask   = pixelate(data, xToPix, yToPix);
+        if is3D
+            set(handle, 'XData', xData(mask));
+            set(handle, 'YData', yData(mask));
+            set(handle, 'ZData', zData(mask));
+        else
+            set(handle, 'XData', data(mask, 1));
+            set(handle, 'YData', data(mask, 2));
+        end
         return
     end
 
     xPixelWidth = 1/xToPix;
     yPixelWidth = 1/yToPix;
-    tol = min(xPixelWidth,yPixelWidth);
+    tol = min(xPixelWidth, yPixelWidth);
 
     % Split up lines which are seperated by NaNs
-    inan   = isnan(vxData) | isnan(vyData);
-    df     = diff([false, ~inan, false]);
-    pstart = find(df == 1);
-    pend   = find(df == -1)-1;
-    nlines = numel(pstart);
-
-    [linesx, linesy] = deal(cell(1,nlines*2));
-    for ii = 1:nlines
-        % Visual data used for simplifications
-        vx = vxData(pstart(ii):pend(ii));
-        vy = vyData(pstart(ii):pend(ii));
-
-        % Actual data that inherits the simplifications
-        x = xData(pstart(ii):pend(ii));
-        y = yData(pstart(ii):pend(ii));
-
-        % Line simplification
-        if numel(vx) > 2
-            mask = opheimSimplify(vx,vy,tol);
-            x    = x(mask);
-            y    = y(mask);
-        end
```

Comment:
Hm actually it took me quite some time to understand what this was acutally doing. However i believe it comes down to whether one has written the code or not. 

One reason why I would change it, is that it is more consistent with the way we handle the split in movePointsCloser.